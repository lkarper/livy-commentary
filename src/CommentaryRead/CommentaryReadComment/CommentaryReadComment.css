.CommentaryReadComment__h3 {
    font-family: Linotype Trajanus;
}