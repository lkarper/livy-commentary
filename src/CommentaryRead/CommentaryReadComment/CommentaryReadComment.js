import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import PropTypes from 'prop-types';
import './CommentaryReadComment.css';

const CommentaryReadComment = (props) => {

    const { comment } = props;

    // if (Object.keys(commentArray.links) !== 0) {
    //     Object.keys(commentArray.links).forEach(linkNumber => {
    //         const anchorString = `<a href="${commentArray.links[linkNumber].url}" target="_blank">${commentArray.links[linkNumber].title}</a>`;
    //         text = text.replace(linkNumber, anchorString);
    //     });
    // }

    return (
        <div>
            <h3 className='CommentaryReadComment__h3'><q>{comment.tag}</q></h3>
            <p>{ReactHtmlParser(comment.comment)}</p>
        </div>    
    );
}

CommentaryReadComment.defaultProps = {
    comment: {
        tag: '',
        comment: '',
    },
};

CommentaryReadComment.propTypes = {
    comment: PropTypes.shape({
        tag: PropTypes.string,
        comment: PropTypes.string,
    }),
};

export default CommentaryReadComment;
