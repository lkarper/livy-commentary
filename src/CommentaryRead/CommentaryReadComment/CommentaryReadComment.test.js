import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentaryReadComment from './CommentaryReadComment';

describe('CommentaryReadComment component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <CommentaryReadComment />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<CommentaryReadComment />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
