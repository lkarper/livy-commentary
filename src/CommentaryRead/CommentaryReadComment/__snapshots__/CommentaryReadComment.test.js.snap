// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentaryReadComment component renders the UI as expected 1`] = `
<CommentaryReadComment
  comment={
    Object {
      "comment": "",
      "tag": "",
    }
  }
>
  <div>
    <h3 />
  </div>
</CommentaryReadComment>
`;
